### ISSUE TITLE
What is the meaning of the boolean variable "use_custom" in the template of gauss_time_and_call?

### ISSUE THREAD

**Issue by marcpaterno** [marcpaterno/gpuintegration#7]
Originally commented on by @f1974939505 in [Issue #1]

> What is the meaning of the boolean variable "use_custom" in the template of gauss_time_and_call?

**Comment by isaki001** [comment]
The `use_custom` template variable is utilized in `gaussian_time_and_call` and several other cases. It enables/disables the use of non-default libraries for several operations such as prefix-summation, reduction, min/max evaluation. By default, we use heavily optimized libraries for these operations

This variable is for experimental purposes and not for providing any library selection instrumentation to the user, which is why it is a boolean. 

This boolean variable was used when comparing cuda, oneAPI and Kokkos. The intent was to remove the optimization factor of the third party libraries in the comparison, since we did not expect the optimization level to be on par due to the varying maturity level of the programming models. Instead, we wanted to compare the internal functions of the integration library and for that, we used custom implementations of those routines. 

When we wanted a head-on comparison, we just would leave the `use_custom` variable as `false` which is the default value of this optional parameter.
